### DISCUSSION TITLE
[@types/jquery] event.which is incorrectly marked as deprecated

### DISCUSSION THREAD

**Discussion by bschwarzent** [DefinitelyTyped/DefinitelyTyped#65584] [Issues with a @types package]
At some point, the `which` property on mouse and keyboard events was [marked as deprecated](https://github.com/jquery/api.jquery.com/issues/821) in jQuery. After some discussion, however, it was ["undeprecated" again](https://github.com/jquery/jquery/issues/4755#issuecomment-664501730).

The TypeScript type definitions for jQuery were [updated](https://github.com/jquery/jquery/issues/2337) when the property was deprecated, but were apparently never changed back when the property was undeprecated. This is annoying, because even with the latest version of @types/jquery (3.5.16) all usages of `event.which` are still marked as deprecated by the IDE, generating unnecessary warnings.

Please update the TS type definitions to match the current API of jQuery, i.e. remove all deprecated tags that refer to issue 821.

![](https://github.com/jquery/jquery/assets/8881838/58254957-07af-40de-b688-ebeb7bea3af9)

(Note: I originally [reported](https://github.com/jquery/jquery/issues/5256) this issue at the jQuery project, but was referred here.)

[a comment by an automated account is not shown]

**Comment by TheNr1Guest** [comment]
I just merged a fix for this:

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67162
